src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriverById =
+                         connection.prepareStatement(getDriverRequest)) {
+            getDriverById.setLong(1, id);
+            ResultSet resultSet = getDriverById.executeQuery();
+            Driver driver = null;
+            if (resultSet.next()) {
+                driver = parseDriver(resultSet);
+            }
+            return Optional.ofNullable(driver);
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get driver by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        List<Driver> allDrivers = new ArrayList<>();
+        String getAllDrivers =
+                "SELECT * FROM drivers WHERE is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllDriversStatement =
+                         connection.prepareStatement(getAllDrivers)) {
+            ResultSet resultSet = getAllDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                allDrivers.add(parseDriver(resultSet));
+            }
+            return allDrivers;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get all drivers from DB "
+                    + allDrivers, e);
+        }
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String updateDriverRequest =
+                "UPDATE drivers SET name = ?, license_number = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateDriverStatement =
+                         connection.prepareStatement(updateDriverRequest)) {
+            updateDriverStatement.setString(1, driver.getName());
+            updateDriverStatement.setString(2, driver.getLicenseNumber());
+            updateDriverStatement.setLong(3, driver.getId());
+            updateDriverStatement.executeUpdate();
+            return driver;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update drivers to DB "
+                    + updateDriverRequest, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteDriverRequest =
+                "UPDATE drivers SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteDriverStatement =
+                         connection.prepareStatement(deleteDriverRequest)) {
+            deleteDriverStatement.setLong(1, id);
+            return deleteDriverStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't delete drivers from DB "
+                    + deleteDriverRequest, e);
+        }
+    }
+
+    private static Driver parseDriver(ResultSet resultSet) {
+        try {
```

Comment:
you call this method inside try-catch blocks, so, you do not need try-catch here